src/main/kotlin/lotto/domain/WinningNumbers.kt

```diff
@@ -0,0 +1,15 @@
+package lotto.domain
+
+data class WinningNumbers(
+    val lotto: Lotto
+) {
+    companion object {
+        fun of(numbers: Set<Int>): WinningNumbers {
+            return try {
+                WinningNumbers(Lotto.of(numbers))
+            } catch (e: IllegalArgumentException) {
+                throw IllegalArgumentException("당첨 번호가 유효하지 않습니다. $numbers")
+            }
```

Comment:
```suggestion
            return WinningNumbers(Lotto.of(numbers))
            
```
여기서 `IllegalArgumentException`이 언제 일어나는지에 대해서 찾아봤는데, 결국 `Lotto`를 만들어줄 때 체크를 하더군요! 차라리 그 체크 로직에다가 에러 메시지를 정의하여 에러가 일어날 때 해당 메시지를 보여주는 게 좋지 않을까요?

Reply:
사실 저렇게 `try catch`로 감싼 이유가 당첨 번호가 로또로 변환된다는 것을 사용자에게 숨기고 싶어서였어요. 유효하지 않은 당첨 번호가 들어오면 `Lotto`에서 생성하는 예외(로또 생성에 대한)가 그대로 나가는 것을 막고, 한 단계 더 추상화된 예외(당첨 번호에 대한)를 보여주고 싶었습니다..!

Reply:
아아! 그렇군요! 그런 의미라면, 저도 좋을 것 같네요😊 여기서 추가한다면 만약 에러 메시지가 번호 중에 어떤 번호가 틀렸는지?도 알려주는 것도 나쁘지는 않을 것 같네요! 🤔